packages/dds/tree/src/simple-tree/api/discrepancies.ts

```diff
@@ -219,52 +176,20 @@ function getViewNodeSchemaType(schema: TreeNodeSchema): SchemaFactoryNodeKind {
  * It is up to the caller to determine whether such discrepancies matter.
  */
 export function* getAllowedContentDiscrepancies(
-	view: FieldSchema,
+	view: TreeSchema,
 	stored: TreeStoredSchema,
 ): Iterable<Discrepancy> {
 	// check root schema discrepancies
-	yield* getFieldDiscrepancies(view, stored.rootFieldSchema, undefined, undefined);
-
-	const viewNodeSchema = new Map<TreeNodeSchemaIdentifier, TreeNodeSchema>();
-
-	walkFieldSchema(view, {
-		node: (schema) => {
-			const identifier: TreeNodeSchemaIdentifier = brand(schema.identifier);
+	yield* getFieldDiscrepancies(view.root, stored.rootFieldSchema, undefined, undefined);
 
-			debugAssert(() => !viewNodeSchema.has(identifier));
-			viewNodeSchema.set(identifier, schema);
-		},
-	});
-
-	for (const [identifier, viewSchema] of viewNodeSchema) {
-		const storedSchema = stored.nodeSchema.get(identifier);
+	for (const [identifier, storedSchema] of stored.nodeSchema) {
+		const viewSchema = view.definitions.get(identifier);
 
 		// if the view schema has a node that's not in the stored schema
-		if (storedSchema === undefined) {
-			const viewType = getViewNodeSchemaType(viewSchema);
-			// TODO does it make sense to have this mismatch when there will also be an allowedTypes mismatch?
```

Comment:
I decided it does not. The stored exist but not view actually is import for can upgrade due to detached fields, but we no longer use this method for computing can update, so I was able to remove this and the below case.